### PR TITLE
Add `diff` and `unwrap`

### PR DESCRIPTION
**Pull request by toslunar**
Takeover #1495.

### CODE REVIEW

File: cupy/math/trigonometric.py [hunk 1]
```diff
@@ -99,7 +103,34 @@
...
+    .. seealso:: :func:`numpy.unwrap`
+    """
+
+    p = cupy.asarray(p)
+    nd = p.ndim
+    dd = sumprod.diff(p, axis=axis)
+    slice1 = [slice(None, None)]*nd     # full slices
+    slice1[axis] = slice(1, None)
+    slice1 = tuple(slice1)
+    ddmod = cupy.mod(dd + numpy.pi, 2*numpy.pi) - numpy.pi
```
Comment (okuta) [inline cupy/math/trigonometric.py]: How about using fusion?

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by grlee77** [comment]
Thanks for adding this.

NumPy 1.16 [added append and prepend arguments to diff](https://github.com/numpy/numpy/blob/v1.16.0/numpy/lib/function_base.py#L1149-L1173). I don't personally need that functionality, but perhaps worth considering for completeness?

**Comment by toslunar** [comment]
> NumPy 1.16 [added append and prepend arguments to diff](https://github.com/numpy/numpy/blob/v1.16.0/numpy/lib/function_base.py#L1149-L1173). I don't personally need that functionality, but perhaps worth considering for completeness?

Done.  Thanks for the suggestion, @grlee77.

**Comment by toslunar** [comment]
The PR uses #1983.

**Comment by okuta** [comment]
pfnci please

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by toslunar** [comment]
> I got error.
> 
> ```
> E   AttributeError: module 'numpy' has no attribute '_NoValue'```
> ```

Thanks.  I'll fix it by using `None`, because CuPy does not support arrays of objects (typecode `'O'`).

**Comment by okuta** [comment]
Thnaks,
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**